Source/Scene/MetadataClassProperty.js

```diff
@@ -562,7 +562,15 @@ function parseType(property, enums) {
  * @private
  */
 MetadataClassProperty.prototype.normalize = function (value) {
-  return normalize(this, value, MetadataComponentType.normalize);
+  if (!this._normalized) {
+    return value;
+  }
+
+  const valueType = this._valueType;
+  const normalizeFunction = function (x) {
+    return MetadataComponentType.normalize(x, valueType);
+  };
+  return transformInPlace(value, normalizeFunction);
```

Comment:
Is it possible to avoid creating a closure like this? The "function pointer" approach from before might be more performant.

Reply:
Or... not worry about it right now since it the alternative feels complicated

Reply:
Options we considered:
1. precompute the closure in MetadataClassProperty  to avoid allocation. though this doesn't change the nested function calls.
2. a lookup table by component type. This would create a lot of boilerplate code and still be closures
3. use `callback.call(undefined, value, ...args)` but this requires ES6
4. use `callback.apply(undefined, [x, otherArgs])` but this causes more allocation...